### PR TITLE
Reuse github action preview

### LINKED ISSUE #2398: Create a reusable GitHub Action to deploy to IBM Container Registry
Currently we have a couple of projects that deploy to IBM Code Engine and use the IBM Container Registry.

For this, we're duplicating some code for the GitHub CI/CD workflows in each repository, which could be reused if available as a GitHub Action.

We should DRY this by creating a reusable GitHub Action.

### PR DESCRIPTION
**Pull request by Tansito**
This PR tries to reuse the new github action that builds and push a docker image to IBM container registry. We are externalising it to improve the reusable logic between projects and interoperability.

Fix #2398 
Related with #2395

### CODE REVIEW

**Comment by Tansito** [comment]
Don't merge, the github action from `gh-actions` is failing.

**Comment by Tansito** [comment]
Waiting for #2400 to fix the problem with the preview action.

**Comment by vabarbosa** [comment]
#2400 #2401 have been merged and preview build failure should no longer be a blocker

File: .github/workflows/preview.yml [hunk 1]
```diff
@@ -7,100 +7,16 @@ jobs:
...
+    uses: Qiskit/gh-actions/.github/workflows/code-engine-preview.yml@main
+    with:
+      container_registry_namespace: community-digital
+      code_engine_project: qiskit.org-preview
+      code_engine_registry_secret: ibm-container-registry
+      docker_image_name: qiskit-org
+      docker_image_port: "3000"
+      ibmcloud_region: us-south
+      ibmcloud_resource_group: Quantum Community
+      pr_tag: pr-${{ github.event.pull_request.number }}
```
Comment (abdonrd) [inline .github/workflows/preview.yml:19]: Now you can remove this line!
Reference: https://github.com/Qiskit/gh-actions/pull/2

**Review by abdonrd — APPROVED** [review]
Yay! 👏🏻

**Review by vabarbosa — APPROVED** [review]
LGTM!
thank you.

**Review by techtolentino — APPROVED** [review]
LGTM 🚀

**Pull request merged**